evalml/data_checks/target_leakage_data_check.py

```diff
@@ -63,14 +65,28 @@ def validate(self, X, y):
         y = _convert_to_woodwork_structure(y)
         X = _convert_woodwork_types_wrapper(X.to_dataframe())
         y = _convert_woodwork_types_wrapper(y.to_series())
+        if not self.pearson:
```

Comment:
Can we extract this into its own method? We could have a `_calculate_pearson` and `_calculate_mutual_information` in this class or something.